docs-guides/source/opt-joint-compression.md

```diff
@@ -0,0 +1,335 @@
+techniques to the weights/activations of the model independently. 
+In this section, we describe how these techniques can be combined, 
+which may be beneficial to get even more disk savings and latency improvements. 
+
+We first start by looking at how to take an uncompressed `mlpackage` 
+and get a joint compressed model by using the `ct.optimize.coreml.*` APIs. 
+As discussed in previous sections as well, this approach may or 
+may not yield a highly accurate model in some cases, however, 
+this is the best way to get the model in the desired format to test 
+out the expected disk size savings and  performance  (latency, runtime memory etc). 
+And once a model is obtained with desired performance characteristics, 
+a better accuracy model can be generated by applying the various data 
+based optimization methods available in `ct.optimize.torch.*`. 
+The latter is discussed via a few API code snippets in the section below. 
+
+## Combining compression types on an mlpackage 
+
+### Joint palettization and quantization 
+
+This means using a Look up table (LUT) whose values are of the 
+dtype INT8/UINT8 instead of Float16 which is the default. 
+This could be beneficial when combined with INT8 activations in
+speeding up inference. e.g. you could take a A16W16 model, 
+quantize the activations to get A8W16 model, and then quantize the 
+weights to a 4 bit LUT, with INT16 dtype, this will yield an A8W4 model, 
+where “W4” refers to a palettized weights with 8 bit LUT. 
+When such a model is run on the NE (on newer SoCs >= A17pro, M4), 
+it will utilize the faster int8-int8 compute path. 
+
+```python
+from coremltools.optimize.coreml import (
+   OptimizationConfig,
+   OpPalettizerConfig,
+   OpLinearQuantizerConfig,
+   palettize_weights,
+   linear_quantize_weights,
+ ) 
+     
+# mlmodel: an uncompressed mlpackage, loaded into memory 
+                                                                          
+# first palettize the model
+# this will produce an LUT with Float values
+op_config = OpPalettizerConfig(nbits=4)
+config = OptimizationConfig(global_config=op_config)
+mlmodel_palettized = palettize_weights(mlmodel, config)
+
+# now apply weight quantization on the model, 
+# with "joint_compression" set to True. 
+# this will result in quantizing the LUT to 8 bits. 
+# (granularity must be set to "per-tensor" for this scenario) 
+op_config = OpLinearQuantizerConfig(mode="linear_symmetric",  
+                                    granularity="per_tensor")
+linear_weight_quantize_config = OptimizationConfig(global_config=op_config)
+
+mlmodel_palettized_with_8bit_lut = linear_quantize_weights(mlmodel_palettized, 
+                                                           linear_weight_quantize_config, 
+                                                           joint_compression=True)
+```
+
+### Joint sparsity and quantization
+
+This means quantizing the non-zero values in the sparse weight tensor to INT8/UINT8 values. 
```

Comment:
Technically could also be INT4/UINT4. Could we say "quantizing the non-zero values in the sparse weight tensor to lower-bit integer values."?

Reply:
I think hardware only supports int8/uint8 currently?

Reply:
Hmm, I think INT4 may also work, but I have not tested it. For now, keeping it INT8 only.